controllers/argocd/util.go

```diff
@@ -1688,3 +1688,31 @@ func UseServer(name string, cr *argoproj.ArgoCD) bool {
 	}
 	return true
 }
+
+// addKubernetesData checks for any Kubernetes-specific labels or annotations
+// in the live object and updates the source object to ensure critical metadata
+// (like scheduling, topology, or lifecycle information) is retained.
+// This helps avoid loss of important Kubernetes-managed metadata during updates.
+func addKubernetesData(source map[string]string, live map[string]string) map[string]string {
```

Comment:
Do we have to return the source map ? the map is already modified inplace in this function isn't it?

Reply:
Forgot golang basics, map are passed by reference😅. Thanks. Refactored the logic.